### PR TITLE
Improve Current Selection in New Find View

### PR DESCRIPTION
**Pull request by Delphinus8821**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [x] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
Updated the current selection drop down list in the new find view and split it into two drop downs, search in and post results action. The search in drop down shows where the search will be performed, e.g. the current selection, current graph or all open graphs. The post search actions drop down has the option to choose whether the search will replace the current selection, add to or remove from the current selection. 

A delete results button was also added to delete results from the graph quickly. It only becomes available after a find all search as it will delete all results found. After pressing the delete results button, the user is then displayed a dialog box which shows how many results will be deleted across x graphs and whether they wish to continue deleting these. 
The delete results option was originally going to be added as part of the Post Search Actions drop down but it was discovered that unless the search with the same results had already been made using a different option in the Post Search Actions drop down, completing the delete results option would cause Constellation to become unresponsive. 

These new options appear in both the basic find and the advanced find. 

### Alternate Designs
N/A

### Why Should This Be In Core?
Improves the usability of the new find view by expanding the different options. Also adds new functionality of being able to delete results from within the find view. 

### Benefits
Easier use of the new find view. 

### Verification Process
Tested all options in both drop down lists on both the basic find and the advanced find to ensure all functionality still works as intended. 
Also tested the delete option on both tabs. 

### Applicable Issues
#1695

### CODE REVIEW

File: CoreFindView/src/au/gov/asd/tac/constellation/views/find2/components/BasicFindTab.java [hunk 1]
```diff
@@ -140,19 +143,29 @@ public BasicFindTab(final FindViewTabs parentComponent) {
...
+        searchInChoiceBox.getSelectionModel().selectedItemProperty().addListener(new ChangeListener<String>() {
             @Override
             public void changed(ObservableValue<? extends String> observableValue, String oldElement, String newElement) {
                 updateSelectionFactors();
+                updateBasicFindParamters();
+            }
+        });
+
+        // set the action for changing the seleciton in the postSearchChoiceBox
+        postSearchChoiceBox.getSelectionModel().selectedItemProperty().addListener(new ChangeListener<String>() {
```
Comment (antares1470) [inline CoreFindView/src/au/gov/asd/tac/constellation/views/find2/components/BasicFindTab.java:156]: can the listener here and the one above be converted to a lambda? Or is there an issue with doing that? (If there's a problem, make the parameters final in the function)

File: CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/BasicFindPlugin.java [hunk 2]
```diff
@@ -196,8 +194,10 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
         foundResult.clear();
         foundResult.addAll(distinctValues);
 
+        // If the results list is null, has different parameters to the current list, and is not searching all graphs, then create a new results list
         if (ActiveFindResultsList.getBasicResultsList() == null || !ActiveFindResultsList.getBasicResultsList().getSearchParameters().equals(this.parameters)
-                || (!this.parameters.isSearchAllGraphs() && ActiveFindResultsList.getBasicResultsList().get(0) != null && !ActiveFindResultsList.getBasicResultsList().get(0).getGraphId().equals(graph.getId()))) {
+                || (!this.parameters.isSearchAllGraphs() && ActiveFindResultsList.getBasicResultsList() == null && ActiveFindResultsList.getBasicResultsList().get(0) != null
```
Comment (antares1470) [inline CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/BasicFindPlugin.java:199]: `ActiveFindResultsList.getBasicResultsList() == null && ActiveFindResultsList.getBasicResultsList().get(0) != null` would appear to cause an issue (if the 1st is true, the 2nd will throw a null pointer)

File: CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java [hunk 3]
```diff
@@ -269,8 +271,10 @@ protected void edit(GraphWriteMethods graph, PluginInteraction interaction, Plug
         foundResult.clear();
         foundResult.addAll(distinctValues);
 
+        // If the results list is null, has different parameters to the current list, and is not searching all graphs, then create a new results list 
         if (ActiveFindResultsList.getAdvancedResultsList() == null || !ActiveFindResultsList.getAdvancedResultsList().getAdvancedSearchParameters().equals(this.parameters)
-                || (!this.parameters.isSearchAllGraphs() && ActiveFindResultsList.getAdvancedResultsList().get(0) != null && !ActiveFindResultsList.getAdvancedResultsList().get(0).getGraphId().equals(graph.getId()))) {
+                || (!this.parameters.getSearchInLocation().equals(ALL_OPEN_GRAPHS) && ActiveFindResultsList.getAdvancedResultsList() == null
+                && ActiveFindResultsList.getAdvancedResultsList().get(0) != null && !ActiveFindResultsList.getAdvancedResultsList().get(0).getGraphId().equals(graph.getId()))) {
```
Comment (antares1470) [inline CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java:277]: Similar here

**Review by antares1470 — CHANGES_REQUESTED** [review]
Some other things from testing:
- When the post-action option is Replace Selection, Find All doesn't highlight any of the found terms and Find Next and Previous append to the selection
- In the Replace tab I'm a bit confused to what "Replace In" means. Replace in this case also seems to have 2 different meanings in this tab (with respect to text and with respect to selection) which I think might be confusing
- I would possibly look at the Delete Results button being a Post-search action option instead. I do like the inclusion of the prompt to check beforehand but I think it doesn't really make sense to be a separate button.
- The term "Delete Results" can also be a little confusing. When I first saw it, I interpreted it as clearing the search rather than deleting things from the graph

**Review by antares1470 — CHANGES_REQUESTED** [review]
Given there is only one option now for the Post actions in the Replace tab, is it possible for that dropdown to be removed?

**Review by antares1470 — APPROVED** [review]

**Comment by mimosa2** [comment]
I tested Basic Find with All Open Graphs and Find All picked up all the hits but I had the default Post-Search Action option as Add To Selection. This option didn't enable Find Prev and Find Next although there were results in the other graphs (they all high lighted correctly).
 I had the similar situation the when the default was set to Remove from Selection.  Is it possible to make no default option for this drop down ( Post-Search Action)?

**Comment by mimosa2** [comment]
After discussed with @Delphinus8821 and @antares1470  I understood that The Find Next & Previous are not meant to be enabled for Add To or Remove From options.

**Review by mimosa2 — APPROVED** [review]

**Pull request merged**